runtime/src/utils/CurriedProviders.js

```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export const CurriedProviders = ({ providers, children }) =>
+    Providers.reduceRight(
+        (acc, Provider) => <Provider>{acc}</Provider>,
+        children
+    )
+CurriedProviders.propTypes = {
+    providers: PropTypes.arrayOf(PropTypes.node.isRequired).isRequired,
```

Comment:
If we were to keep this implementation, a provider would be a function or a class, not a node. It only becomes a react node once we render it as a `<Provider/>`

Reply:
Correct, good catch